src/coreclr/clrfeatures.cmake

```diff
@@ -38,4 +38,20 @@ endif()
 
 if (CLR_CMAKE_TARGET_WIN32)
   set(FEATURE_TYPEEQUIVALENCE 1)
-endif(CLR_CMAKE_TARGET_WIN32)
\ No newline at end of file
+endif(CLR_CMAKE_TARGET_WIN32)
+
+# Enable cached interface dispatch so that we can test/debug it more easily on non-embedded scenarios (set DOTNET_UseCachedInterfaceDispatch=1)
+# Only enable in chk/debug builds as this support isn't intended for retail use elsewhere
+if (((CMAKE_BUILD_TYPE STREQUAL "Debug") OR (CMAKE_BUILD_TYPE STREQUAL "Checked")) AND (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64))
```

Comment:
This won't work when cmake uses multi-configuration target generator, like the one for msbuild on windows (ninja doesn't have this problem). The cmake generates multi-configuration build files that contain support for both Debug, Checked and Release in one pass. So the `CMAKE_BUILD_TYPE` is empty. 
The way to handle that properly is to use generator expressions.
So in this case, it would be
```cmake
set(FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH $<IF:$<CONFIG:Debug,Checked>,1,0>)
```
and then at the place where we set the related define, we would have
```cmake
add_compile_definitions($<${FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH}:FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH>)
```

Reply:
I've updated to support the generator expressions for this.